### ISSUE TITLE
References to SUPER data types in Redshift From Clauses are incorrectly flagged

### ISSUE THREAD

**Issue by PLBMR** [sqlfluff/sqlfluff#2567]
PR #2564 will fix all false rule violations in Issue #1672 except for:
```bash
L:   3 | P:   5 | L027 | Unqualified reference 'o' found in select with more than one referenced table/view.
L:   4 | P:  36 | L025 | Alias 'o' is never used in SELECT statement.
```

@PLBMR will address these false rule violations in this issue.

_Originally posted by @PLBMR in https://github.com/sqlfluff/sqlfluff/issues/1672#issuecomment-1030410954_

**Comment by barrywhart** [comment]
@PLBMR: I notice that your recently merged PR says it "partially fixes" this issue. Can you please update the issue to reflect the work that remains to be done?

**Comment by PLBMR** [comment]
Issue is reproduced after release `sqlfluff==0.10.0`:

```bash
mkdir -p ~/.virtualenvs
python3.8 -m venv ~/.virtualenvs/issue-2567-reproduce
source ~/.virtualenvs/issue-2567-reproduce/bin/activate
pip install -r requirements.txt
...
sqlfluff lint --dialect=redshift super_select.sql # see issue #1672 for super_select.sql definition
== [super_select.sql] FAIL                                                                                                                                                        
L:   3 | P:   5 | L027 | Unqualified reference 'o' found in select with more than one referenced table/view.
L:   4 | P:  36 | L025 | Alias 'o' is never used in SELECT statement.
All Finished 📜 🎉!
```

I'll start working on fixing this.

**Comment by barrywhart** [comment]
@PLBMR: Can you provide an example query for the issue being described here? Issues are easier to understand, work on, and review if they are fairly self contained.

**Comment by PLBMR** [comment]
@barrywhart the false rule violation described above, and the reproduction of the false rule violation, are both identified on this query from issue #1672 :

```sql
SELECT
    c[0].col,
    o
FROM customer_orders c, c.c_orders o;
```

The way we can fix this is by allowing TableReferenceSegments in the redshift dialect to also be references to `SUPER` fields. I'm working on the fix for this in my own branch.

**Comment by PLBMR** [comment]
Issue still exists as of version `0.10.1`. Will address today.
